From b6749f4b606470bec455b4d0bd1f15d8667dea3c Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Mon, 24 Jan 2011 17:39:45 +0000
Subject: BluetoothStatus: depend more on libgnome-bluetooth-applet

Ensure that a dependency is generated even when using --as-needed,
by adding a fake function that calls into the library.

https://bugzilla.gnome.org/show_bug.cgi?id=639324
---
diff --git a/src/gnome-shell-plugin.c b/src/gnome-shell-plugin.c
index 96075f4..8d85fa2 100644
--- a/src/gnome-shell-plugin.c
+++ b/src/gnome-shell-plugin.c
@@ -670,3 +670,17 @@ MetaPluginInfo *gnome_shell_plugin_plugin_info (MetaPlugin *plugin)
 
   return &info;
 }
+
+#if HAVE_BLUETOOTH
+/* HACK:
+   Add a non-static function that calls into libgnome-bluetooth-applet.so,
+   to avoid the linker being too smart and removing the dependency.
+   This function is never actually called.
+*/
+extern GType bluetooth_applet_get_type(void);
+void _shell_link_to_bluetooth(void);
+
+void _shell_link_to_bluetooth(void) {
+  bluetooth_applet_get_type();
+}
+#endif
--
cgit v0.8.3.4
